### PR TITLE
Refactor ReplAutocompletion

### PR DESCRIPTION
**Pull request by rexim**
Split #306 

### Remaining Work

- Extract and cover with tests
    - `object CommandWithArg`
    - `def matchingDefinitions()`
    - `def lastNameInLine()`
    - `def matches()`
- Reimplement `autocompleteLoadStatement` with Morganey index

### CODE REVIEW

File: src/main/scala/me/rexim/morganey/autocompletion/extractors/LoadStatement.scala [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    val pathElements = path.split("\\.", -1).toList
+    val endsWithDot  = pathElements.lastOption.exists(_.isEmpty)
+    val realPathElements =
+      if (endsWithDot) pathElements.init
+      else pathElements
+    (realPathElements, endsWithDot)
+  }
+
+  private def handleLoading(load: MorganeyLoading) =
+    load.modulePath map pathInformation getOrElse zeroLoad
```
Comment (rexim) [inline src/main/scala/me/rexim/morganey/autocompletion/extractors/LoadStatement.scala:20]: This is the only place where `zeroLoad` is used. Can be inlined.

Comment (ForNeVeR) [reply src/main/scala/me/rexim/morganey/autocompletion/extractors/LoadStatement.scala:20]: Not sure if you need to.

**Review by ForNeVeR — APPROVED** [review]

**Review by keddelzz — APPROVED** [review]
LGTM 👍

**Comment by rexim** [comment]
@ForNeVeR @keddelzz cool! Thank you guys for the review! :)

**Comment by rexim** [comment]
I think #216 is the deepest split issue ever! We desperately need @Issuestant and its ability to visualize split forest https://github.com/tsoding/Issuestant/issues/18

**Pull request merged**